processor/src/main/java/org/mapstruct/ap/internal/model/LifecycleMethodResolver.java

```diff
@@ -91,7 +91,11 @@ private LifecycleMethodResolver() {
         List<SourceMethod> availableMethods =
             new ArrayList<SourceMethod>( methodsProvidedByParams.size() + sourceModelMethods.size() );
 
-        availableMethods.addAll( methodsProvidedByParams );
+        for ( SourceMethod methodProvidedByParams : methodsProvidedByParams ) {
+            if ( methodProvidedByParams.isLifecycleCallbackMethod() ) {
```

Comment:
this looks like a task of the `MethodFamiliySelector`. Need to analyse whether to do it here or leave it up to this selector.